### DISCUSSION TITLE
I2C error between AI-deck and BMP280

### DISCUSSION THREAD

**Discussion by mfxjx000205** [bitcraze/discussions#1212] [Q&A]
Hi, Bitcraze,

In Question #1172 , I asked you about the I2C driver not working on the AI-deck. After the Lunar New Year, I double-checked the schematic and found that the I2C interface was indeed missing the pull-up resistor. However, I have tried repeatedly and still cannot get the AI-deck to drive the I2C device correctly.

Below, I will explain a few facts to you:

1. I configured the pull-up resistors for the pins on the header in exactly the same way as the I2C pins on the camera on the AI-deck. The sensor I chose is a bmp280, the full example is given in [gap_sdk](https://github.com/GreenWaves-Technologies/gap_sdk/blob/master/examples/gap8/basic/interfaces/i2c/i2c_bmp280/test_i2c_bmp280.c), I just replaced the I2C address in the example with my bmp280 address: 0x76. After running the code, unfortunately, pi_i2c_read( ) still only reads 0xFF.
![20240302-212339](https://github.com/bitcraze/discussions/assets/32474853/dd5cab0c-8776-4f54-9f1c-8d35c5c86fdb)
![20240302-212346](https://github.com/bitcraze/discussions/assets/32474853/07bf9fea-8289-469c-8d92-1cb0717c9d47)

2. I tested this on a raspberry pi zero 2w and the bmp280 I purchased is fully detectable while manipulating the registers.

Since I2C is a relatively simple protocol and circuit, I don't think my understanding of I2C should be wrong. Therefore, I have the following two suspicions:

1. Does the GPIO ports B34 and D1 on the GAP8 chip require some kind of additional configuration to turn on the I2C function.
2. The two GPIO ports, B34 and D1, of the GAP8 chip, have sufficient drive capability, or, in other words, are they capable of conducting the internal MOS tubes.

Next week, I'll use an oscilloscope to probe for problems such as weak drive on the comms link, as it's difficult to grab such waveforms with an oscilloscope and I'm not very proficient at it, so this will delay things a bit.

I would appreciate any content that might be helpful to me.

Best Wishes,

Hanjie

**Comment by tobbeanton** [comment]
I've not used the I2C on the GAP8 myself and question would probably be answered better using their support forum but my initial thought is that you setup should work and that it is most likely a configuration problem.
